pkg/operator/sync.go

```diff
@@ -212,92 +222,64 @@ func (optr *Operator) syncMutatingWebhook() error {
-			klog.Error(lastError)
-			return false, nil
-		}
+func (optr *Operator) checkDeploymentRolloutStatus(resource *appsv1.Deployment, requeuePeriod time.Duration) (bool, time.Duration, error) {
+	d, err := optr.kubeClient.AppsV1().Deployments(resource.Namespace).Get(context.Background(), resource.Name, metav1.GetOptions{})
+	if apierrors.IsNotFound(err) {
+		return false, 0, fmt.Errorf("deployment %s is not found", resource.Name)
+	}
+	if err != nil {
+		return false, 0, fmt.Errorf("getting Deployment %s during rollout: %v", resource.Name, err)
+	}
 
-		if d.DeletionTimestamp != nil {
-			lastError = nil
-			return false, fmt.Errorf("deployment %s is being deleted", resource.Name)
-		}
+	if d.DeletionTimestamp != nil {
+		return false, 0, fmt.Errorf("deployment %s is being deleted", resource.Name)
+	}
 
-		if d.Generation <= d.Status.ObservedGeneration && d.Status.UpdatedReplicas == d.Status.Replicas && d.Status.UnavailableReplicas == 0 {
-			c := conditions.GetDeploymentCondition(d, appsv1.DeploymentAvailable)
-			if c == nil {
-				lastError = fmt.Errorf("deployment %s is not reporting available yet", resource.Name)
-				klog.V(4).Info(lastError)
-				return false, nil
-			}
-			if c.Status == corev1.ConditionFalse {
-				lastError = fmt.Errorf("deployment %s is reporting available=false", resource.Name)
-				klog.V(4).Info(lastError)
-				return false, nil
-			}
-			if c.LastTransitionTime.Time.Add(deploymentMinimumAvailabilityTime).After(time.Now()) {
-				lastError = fmt.Errorf("deployment %s has been available for less than 3 min", resource.Name)
-				klog.V(4).Info(lastError)
-				return false, nil
-			}
-
-			lastError = nil
-			return true, nil
-		}
+	if d.Generation > d.Status.ObservedGeneration || d.Status.UpdatedReplicas != d.Status.Replicas || d.Status.UnavailableReplicas > 0 {
+		klog.V(3).Infof("deployment %s is not ready. status: (replicas: %d, updated: %d, ready: %d, unavailable: %d)", d.Name, d.Status.Replicas, d.Status.UpdatedReplicas, d.Status.ReadyReplicas, d.Status.UnavailableReplicas)
+		return true, requeuePeriod, nil
+	}
+
+	c := conditions.GetDeploymentCondition(d, appsv1.DeploymentAvailable)
+	if c == nil {
+		klog.V(3).Infof("deployment %s is not reporting available yet", resource.Name)
+		return true, requeuePeriod, nil
+	}
 
-		lastError = fmt.Errorf("deployment %s is not ready. status: (replicas: %d, updated: %d, ready: %d, unavailable: %d)", d.Name, d.Status.Replicas, d.Status.UpdatedReplicas, d.Status.ReadyReplicas, d.Status.UnavailableReplicas)
-		klog.V(4).Info(lastError)
-		return false, nil
-	})
-	if lastError != nil {
-		return lastError
+	if c.Status == corev1.ConditionFalse {
+		klog.V(3).Infof("deployment %s is reporting available=false", resource.Name)
+		return true, requeuePeriod, nil
 	}
-	return err
+
+	if c.LastTransitionTime.Time.Add(deploymentMinimumAvailabilityTime).After(time.Now()) {
+		klog.V(3).Infof("deployment %s has been available for less than 3 min", resource.Name)
```

Comment:
From where `3 min` did come?

Reply:
3 min is the value of the `deploymentMinimumAvailabilityTime`, I have just copied this string from the previous code, this isn't new

Reply:
Can template log message with variable value? :)